### DISCUSSION TITLE
Macro Accuracy vs Micro Accuracy

### DISCUSSION THREAD

**Discussion by camilomarino** [Lightning-AI/torchmetrics#863] [Classification] [answered]
Hello,

Sorry if the question is obvious, but I've been messing around with torchmetrics for a while and haven't understood the result I'm getting.

I have doubts between the difference of macro-accuracy and micro-accuracy. I have found definitions for macro-micro recovery/accuracy on the web, but not for accuracy. In particular, I wouldn't understand what it means to calculate accuracy per class.

For example, with the following binary classification example, I get 2 different results for the micro and macro accuracy:

````py
import torchmetrics
import torch

target     = torch.tensor([1, 1, 0, 0, 0, 0, 0])
prediction = torch.tensor([1, 0, 1, 0, 0, 0, 0])
print("micro:", torchmetrics.functional.precision(target, prediction, "micro", num_classes=2))
print("macro:", torchmetrics.functional.precision(target, prediction, "macro", num_classes=2))
````
````
micro: tensor(0.7143)
macro: tensor(0.6500)
````

The associated confusion matrix is:
![Figure_11](https://user-images.githubusercontent.com/47334227/155855650-26b7ebf5-e340-4500-9541-b26a804ad334.png)

I appreciate if you could clarify it for me.

**Comment by weiaicunzai** [comment, marked as answer]
I believe micro means  overall accuracy, and macro means class-wise accuracy.
```
target     = torch.tensor([1, 1, 0, 0, 0, 0, 0])
prediction = torch.tensor([1, 0, 1, 0, 0, 0, 0])
```
you have 7 samples.  5 of them are predicted  crrectly,  micro = 5 / 7 = 0.714

you have 5 samples with class 0, 80% of class 0 samples are right,   you have 2 samples with class 1, 50% of them is right ,  class-wise average is (50 + 80) / 2 = 130 / 2 = 65%

**Reply by Borda** [reply]
you can also read here: https://datascience.stackexchange.com/questions/15989/micro-average-vs-macro-average-performance-in-a-multiclass-classification-settin
